### PR TITLE
TODOs 06 and 07

### PR DESCRIPTION
**Pull request by arkadiuszpietraszek**
- write about getAttMtu()
  - write about requestAttMtuChange()
  - write about addAttMtuChangeListener()
  - write about removeAttMtuChangeListener()

Signed-off-by: Arkadiusz Pietraszek <a.pietraszek@samsung.com>

### CODE REVIEW

File: docs/application/web/guides/connectivity/bluetooth.md [hunk 1]
```diff
@@ -718,13 +718,112 @@ To read and write a value of the Bluetooth descriptor:
...
+
+1. Get BLE adapter:
+   ```
+   var adapter = tizen.bluetooth.getLEAdapter();
+   ```
+2. Start scannining and search for a device:
+   ```
+   var device;
+   adapter.startScan(function onsuccess(scannedDevice) {
+   if (device.name == "MyBLEDevice") {
```
Comment (djuszczakk) [inline docs/application/web/guides/connectivity/bluetooth.md]:
```suggestion
   if (scannedDevice.name == "MyBLEDevice") {
```

File: docs/application/web/guides/connectivity/bluetooth.md [hunk 2]
```diff
@@ -718,13 +718,112 @@ To read and write a value of the Bluetooth descriptor:
...
```
Comment (djuszczakk) [inline docs/application/web/guides/connectivity/bluetooth.md]:
```suggestion
   if (scannedDevice.name == "MyBLEDevice") {
```

File: docs/application/web/guides/connectivity/bluetooth.md [hunk 3]
```diff
@@ -718,13 +718,112 @@ To read and write a value of the Bluetooth descriptor:
...
+   device.connect(onconnected, onerror);
+   ```
+5. If the device has been connected call `addAttMtuChangeListener()` function providing as a parameter the callback function to be called on each change of the ATT MTU value:
+   ```
+   function attMtuChangeCallback(newAttMtuValue) {
+      console.log("ATT MTU value has changed to: " + newAttMtuValue);
+   }
+   var listenerId = device.addAttMtuChangeListener(attMtuChangeCallback);
+   ```
+   The `listenerId` value stores an identificator of the listener, wchich is needed to remove the listener.
```
Comment (djuszczakk) [inline docs/application/web/guides/connectivity/bluetooth.md]:
```suggestion
   The `listenerId` value stores an identificator of the listener, which is needed to remove the listener.
```

**Review by djuszczakk — CHANGES_REQUESTED** [review]

**Comment by djuszczakk** [comment]
Is there a possibility to change ATT MTU value on the server side?

File: docs/application/web/guides/connectivity/bluetooth.md [hunk 4]
```diff
@@ -718,13 +718,112 @@ To read and write a value of the Bluetooth descriptor:
...
+   ```
+   var attMtu = device.getAttMtu();
+   ```
+6. If the device has been connected call `requestAttMtuChange()` function on the device object to request a change of the ATT MTU value. Pass desired ATT MTU value as an attribute:
+   ```
+   var newAttMtuValue = 64;
+   device.requestAttMtuChange(newAttMtuValue);
+   ```
+   > **Note**
+   > Call to `requestAttMtuChange()` function should be accepted only if both devices support new ATT MTU value.
```
Comment (pwasowski2) [inline docs/application/web/guides/connectivity/bluetooth.md]: "should be accepted" - what does it mean, precisely?

File: docs/application/web/guides/connectivity/bluetooth.md [hunk 5]
```diff
@@ -718,13 +718,112 @@ To read and write a value of the Bluetooth descriptor:
...
+      console.log("Failed to connect to device: " + e.message);
+   }
+
+   function onconnected() {
+      console.log("Connected to device");
+   }
+
+   device.connect(onconnected, onerror);
+   ```
+5. If the device has been connected call `getAttMtu()` function on the device object to get current ATT MTU value:
```
Comment (pwasowski2) [inline docs/application/web/guides/connectivity/bluetooth.md]: I think, that points 1-4 are redundant - the guide (https://docs.tizen.org/application/web/guides/connectivity/bluetooth/#connecting-to-a-bluetooth-low-energy-device) already has a manual, instructing the readers how to establish a connection with a GATT server (https://docs.tizen.org/application/web/guides/connectivity/bluetooth/#connecting-to-a-bluetooth-low-energy-device).

Please, take a look at other sections of the existing manual, e.g. https://docs.tizen.org/application/web/guides/connectivity/bluetooth/#receiving-notifications-on-characteristic-value-changes.
They don't list all the steps required to make a connection, but link to the proper, existing section of the manual:

> 1. [Connect to a Bluetooth Low Energy device.](https://docs.tizen.org/application/web/guides/connectivity/bluetooth/#connecting-to-a-bluetooth-low-energy-device)

Please, consider removing the points 1-4 from this section.

File: docs/application/web/guides/connectivity/bluetooth.md [hunk 6]
```diff
@@ -718,13 +718,112 @@ To read and write a value of the Bluetooth descriptor:
...
+      console.log("Failed to connect to device: " + e.message);
+   }
+
+   function onconnected() {
+      console.log("Connected to device");
+   }
+
+   device.connect(onconnected, onerror);
+   ```
+5. If the device has been connected call `addAttMtuChangeListener()` function providing as a parameter the callback function to be called on each change of the ATT MTU value:
```
Comment (pwasowski2) [inline docs/application/web/guides/connectivity/bluetooth.md]: The same as above - I think points 1-4 are unnecessary and can be replaced with a link to the section, that tells how to establish a connection with a server.

File: docs/application/web/guides/connectivity/bluetooth.md [hunk 7]
```diff
@@ -718,13 +718,112 @@ To read and write a value of the Bluetooth descriptor:
...
+      console.log("Connected to device");
+   }
+
+   device.connect(onconnected, onerror);
+   ```
+5. If the device has been connected call `getAttMtu()` function on the device object to get current ATT MTU value:
+   ```
+   var attMtu = device.getAttMtu();
+   ```
+6. If the device has been connected call `requestAttMtuChange()` function on the device object to request a change of the ATT MTU value. Pass desired ATT MTU value as an attribute:
```
Comment (pwasowski2) [inline docs/application/web/guides/connectivity/bluetooth.md]:
```suggestion
6. If the `device` has been connected call `requestAttMtuChange()` function on the `device` object to request a change of the ATT MTU value. Pass desired ATT MTU value as an argument:
```

File: docs/application/web/guides/connectivity/bluetooth.md (hunk 6 above)
Comment (pwasowski2) [inline docs/application/web/guides/connectivity/bluetooth.md]:
```suggestion
5. If the `device` has been connected call `addAttMtuChangeListener()` function providing as a parameter the callback function to be called on each change of the ATT MTU value:
```

File: docs/application/web/guides/connectivity/bluetooth.md [hunk 8]
```diff
@@ -718,13 +718,112 @@ To read and write a value of the Bluetooth descriptor:
...
+6. After setting up the listener with `addAttMtuChangeListener()` function, the callback can be invoked by changing the ATT MTU value:
+   ```
+   var newAttMtuValue = 50;
+   device.requestAttMtuChange(newAttMtuValue);
+   ```
+   The change of ATT MTU value will trigger the callback function.
+   > **Note**
+   > Call to `requestAttMtuChange()` function should be accepted only if both devices support new ATT MTU value.
+
+When a callback monitoring the ATT MTU value changes is no longer needed, you can remove it:
```
Comment (pwasowski2) [inline docs/application/web/guides/connectivity/bluetooth.md]:
```suggestion
When a listener monitoring the ATT MTU value changes is no longer needed, you can remove it:
```

File: docs/application/web/guides/connectivity/bluetooth.md [hunk 9]
```diff
@@ -718,13 +718,112 @@ To read and write a value of the Bluetooth descriptor:
...
+
+7. Call `removeAttMtuChangeListener()` providing the identificator of the listener you want to remove.
```
Comment (pwasowski2) [inline docs/application/web/guides/connectivity/bluetooth.md]:
```suggestion
7. Call `removeAttMtuChangeListener()` providing the identifier of the listener you want to remove.
```

File: docs/application/web/guides/connectivity/bluetooth.md (hunk 3 above)
Comment (pwasowski2) [inline docs/application/web/guides/connectivity/bluetooth.md]:
```suggestion
   The `listenerId` value stores an identifier of the listener, wchich is needed to remove the listener.
```

Comment (pwasowski2) [inline docs/application/web/guides/connectivity/bluetooth.md]:
```suggestion
   The `listenerId` value stores an identifier of the listener, wchich is needed to remove the listener.
```

**Review by pwasowski2 — CHANGES_REQUESTED** [review]

File: docs/application/web/guides/connectivity/bluetooth.md (hunk 1 above)
Comment (arkadiuszpietraszek) [reply docs/application/web/guides/connectivity/bluetooth.md:732]: Done

**Comment by arkadiuszpietraszek** [comment]
> Is there a possibility to change ATT MTU value on the server side?

No. Native API do not support this.

File: docs/application/web/guides/connectivity/bluetooth.md (hunk 3 above)
Comment (arkadiuszpietraszek) [reply docs/application/web/guides/connectivity/bluetooth.md]: Done

Comment (arkadiuszpietraszek) [reply docs/application/web/guides/connectivity/bluetooth.md]: Done

File: docs/application/web/guides/connectivity/bluetooth.md (hunk 9 above)
Comment (arkadiuszpietraszek) [reply docs/application/web/guides/connectivity/bluetooth.md]: Done

File: docs/application/web/guides/connectivity/bluetooth.md (hunk 8 above)
Comment (arkadiuszpietraszek) [reply docs/application/web/guides/connectivity/bluetooth.md]: Done.

File: docs/application/web/guides/connectivity/bluetooth.md (hunk 6 above)
Comment (arkadiuszpietraszek) [reply docs/application/web/guides/connectivity/bluetooth.md]: Done

File: docs/application/web/guides/connectivity/bluetooth.md (hunk 7 above)
Comment (arkadiuszpietraszek) [reply docs/application/web/guides/connectivity/bluetooth.md]: Done

File: docs/application/web/guides/connectivity/bluetooth.md (hunk 5 above)
Comment (arkadiuszpietraszek) [reply docs/application/web/guides/connectivity/bluetooth.md]: Done

File: docs/application/web/guides/connectivity/bluetooth.md (hunk 6 above)
Comment (arkadiuszpietraszek) [reply docs/application/web/guides/connectivity/bluetooth.md]: Done

File: docs/application/web/guides/connectivity/bluetooth.md (hunk 4 above)
Comment (arkadiuszpietraszek) [reply docs/application/web/guides/connectivity/bluetooth.md]: I rephrased this Note. to:
After calling `requestAttMtuChange()` function ATT MTU value change should be accepted if both devices support new ATT MTU value according to the BLE stanard.

File: docs/application/web/guides/connectivity/bluetooth.md (hunk 3 above)
Comment (arkadiuszpietraszek) [reply docs/application/web/guides/connectivity/bluetooth.md]: Done

File: docs/application/web/guides/connectivity/bluetooth.md (hunk 2 above)
Comment (arkadiuszpietraszek) [reply docs/application/web/guides/connectivity/bluetooth.md]: Removed.

File: docs/application/web/guides/connectivity/bluetooth.md (hunk 1 above)
Comment (arkadiuszpietraszek) [reply docs/application/web/guides/connectivity/bluetooth.md]: Removed.

File: docs/application/web/guides/connectivity/bluetooth.md [hunk 10]
```diff
@@ -718,13 +718,56 @@ To read and write a value of the Bluetooth descriptor:
...
+   > After calling `requestAttMtuChange()` function ATT MTU value change should be accepted if both devices support new ATT MTU value according to the [Bluetooth Core Specification](https://www.bluetooth.com/specifications/bluetooth-core-specification/).
 <!-- END #06 -->
 
 <!-- TODO #07
       - write about addAttMtuChangeListener()
       - write about removeAttMtuChangeListener()
  -->
 ### Receiving Notifications on ATT MTU changes
+
+To recieve notification on ATT MTU value changes:
```
Comment (djuszczakk) [inline docs/application/web/guides/connectivity/bluetooth.md]:
```suggestion
To recieve notifications on ATT MTU value changes:
```

**Review by djuszczakk — CHANGES_REQUESTED** [review]

**Review by djuszczakk — APPROVED** [review]

**Review by djuszczakk — APPROVED** [review]

**Review by pwasowski2 — APPROVED** [review]

**Pull request merged**